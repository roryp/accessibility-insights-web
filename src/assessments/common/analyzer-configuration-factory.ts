// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Messages } from 'common/messages';
import { RuleAnalyzerConfiguration } from 'injected/analyzers/analyzer';
import { ScannerUtils } from 'injected/scanner-utils';

export class AnalyzerConfigurationFactory {
    public static forScanner(base: Partial<RuleAnalyzerConfiguration>): RuleAnalyzerConfiguration {
        const defaultValues: Partial<RuleAnalyzerConfiguration> = {
            resultProcessor: (scanner: ScannerUtils) => scanner.getAllCompletedInstances,
            telemetryProcessor: telemetryFactory => telemetryFactory.forAssessmentRequirementScan,
            analyzerMessageType: Messages.Assessment.AssessmentScanCompleted,
        };

        return {
            ...defaultValues,
            ...base,
        } as RuleAnalyzerConfiguration;
    }
}
