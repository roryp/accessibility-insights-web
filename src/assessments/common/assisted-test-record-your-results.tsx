// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export const AssistedTestRecordYourResults = () => (
    <li>
        Record your results:
        <ol>
            <li>
                Select <Markup.Term>Fail</Markup.Term> for any instances that do not meet the
                requirement.
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>. Or, after you have marked all
                failures, select <Markup.Term>Pass unmarked instances.</Markup.Term>
            </li>
        </ol>
    </li>
);
