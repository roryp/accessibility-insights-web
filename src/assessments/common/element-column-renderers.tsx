// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { AssessmentInstanceDetailsColumn } from 'DetailsView/components/assessment-instance-details-column';
import * as React from 'react';

export function onRenderPathColumn(item: InstanceTableRow): JSX.Element {
    let textContent = '';
    if (item.instance.target) {
        textContent = item.instance.target.join(';');
    }

    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={textContent}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}

export function onRenderSnippetColumn(item: InstanceTableRow): JSX.Element {
    return (
        <AssessmentInstanceDetailsColumn
            background={null}
            labelText={null}
            textContent={item.instance.html}
            tooltipId={null}
            customClassName="not-applicable"
        />
    );
}
