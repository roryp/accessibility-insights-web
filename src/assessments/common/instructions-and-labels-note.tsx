// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import * as React from 'react';

import * as Markup from '../markup';

export const InstructionsAndLabelsNotes = () => (
    <Markup.Emphasis>
        Note: For WCAG 2.1 compliance, both visible instructions and labels must be programmatically
        related to a widget. For WCAG 2.0 compliance, only visible instructions must be
        programmatically related to a widget.
    </Markup.Emphasis>
);
