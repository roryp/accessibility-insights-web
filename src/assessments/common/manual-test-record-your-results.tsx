// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import * as Markup from '../markup';

export interface ManualTestRecordYourResultsProps {
    isMultipleFailurePossible: boolean;
}

export const ManualTestRecordYourResults = (props: ManualTestRecordYourResultsProps) => (
    <li>
        Record your results:
        <ol>
            <li>
                If you find {props.isMultipleFailurePossible ? 'any failures' : 'a failure'}, select{' '}
                <Markup.Term>Fail</Markup.Term>,
                {props.isMultipleFailurePossible
                    ? ' then add them as failure instances'
                    : ' then add the failure instance'}
                .
            </li>
            <li>
                Otherwise, select <Markup.Term>Pass</Markup.Term>.
            </li>
        </ol>
    </li>
);
