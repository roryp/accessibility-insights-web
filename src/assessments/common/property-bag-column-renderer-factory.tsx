// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { AssessmentInstanceRowData } from 'DetailsView/components/assessment-instance-table';
import {
    propertyBagColumnRenderer,
    PropertyBagColumnRendererConfig,
} from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static getRenderer<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: AssessmentInstanceRowData<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
