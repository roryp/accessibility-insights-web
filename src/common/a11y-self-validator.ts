// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ScannerUtils } from '../injected/scanner-utils';
import { ScanResults } from '../scanner/iruleresults';
import { HTMLElementUtils } from './html-element-utils';
import { Logger } from './logging/logger';

export interface LoggedRule {
    id: string;
    description: string;
    nodes: LoggedNode[];
}

export interface LoggedNode {
    target: string[];
    domElement: HTMLElement;
    all: FormattedCheckResult[];
    // tslint:disable-next-line: no-reserved-keywords
    any: FormattedCheckResult[];
    none: FormattedCheckResult[];
}

export class A11YSelfValidator {
    constructor(
        private scannerUtils: ScannerUtils,
        private docUtils: HTMLElementUtils,
        private logger: Logger,
    ) {}

    public validate(): void {
        this.scannerUtils.scan(null, this.logAxeResults);
    }

    private logAxeResults = (axeResults: ScanResults): void => {
        const violations = axeResults.violations;
        const loggedViolations: LoggedRule[] = [];

        if (violations && violations.length > 0) {
            violations.forEach(violation => {
                loggedViolations.push({
                    id: violation.id,
                    description: violation.description,
                    nodes: violation.nodes.map(node => {
                        return {
                            domElement: this.docUtils.querySelector(node.target[0]) as HTMLElement,
                            all: node.all,
                            any: node.any,
                            none: node.none,
                            target: node.target,
                        } as LoggedNode;
                    }),
                });
            });
        }

        this.logger.log(loggedViolations);
    };
}
