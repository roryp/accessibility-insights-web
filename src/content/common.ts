// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { ContentCreator } from 'views/content/content-page';
import { GuidanceTitle as gt } from 'views/content/guidance-title';
import { link } from './link';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
