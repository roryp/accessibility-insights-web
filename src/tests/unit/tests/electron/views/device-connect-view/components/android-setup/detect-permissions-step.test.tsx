// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DetectPermissionsStep } from 'electron/views/device-connect-view/components/android-setup/detect-permissions-step';
import { shallow } from 'enzyme';
import * as React from 'react';
import { AndroidSetupStepPropsBuilder } from 'tests/unit/common/android-setup-step-props-builder';

describe('DetectDevicesStep', () => {
    const props = new AndroidSetupStepPropsBuilder('detect-permissions').build();

    it('renders', () => {
        const rendered = shallow(<DetectPermissionsStep {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });
});
