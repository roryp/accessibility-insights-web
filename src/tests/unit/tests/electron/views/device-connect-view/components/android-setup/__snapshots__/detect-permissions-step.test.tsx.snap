// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetectDevicesStep renders 1`] = `
<AndroidSetupSpinnerStep
  deps={
    Object {
      "LinkComponent": [Function],
    }
  }
  disableCancel={true}
  spinnerLabel="Checking permissions..."
/>
`;
